In [1]:
from math import sqrt

## Урок 5. Применение математической статистики для проверки гипотез в реальной жизни для популярных метрик

### Задание 1  
Вы провели эксперимент c упрощением формы заказа в магазине Утконос и получили результаты по метрике конверсий в покупку. Выберите метод оценки и оцените есть ли стат.значимые различия между конверсиями в двух группах при alpha = 5%. Дайте краткие рекомендации команде.  
Результаты:
1) Число юзеров в группах , которые заходили на сайт в период эксперимента: N1 = 15550 и N2 = 15550.  
2) Число юзеров в группах , которые совершили хотя бы одну покупку за период эксперимента: n1 = 164 и n2 = 228   
3) Конверсии : conv1 = 1.05% conv2 = 1.47%.

#### Решение:  
Для оценки статистически значимых различий между конверсиями в двух группах при α = 5% используем Z критерий.

Z = (conv1 - p2) / sqrt(p * (1 - p) * (1/N1 + 1/N2))

где p = (n1 * conv1 + n2 * conv2) / (n1 + n2)

In [7]:
conv1 = 1.05 / 100
conv2 = 1.47 / 100
n1 = 164
n2 = 228
N1 = N2 = 15550

p =  (n1 * conv1 + n2 * conv2) / (n1 + n2)
z = (conv1 - conv2) / sqrt(p * (1 - p) * (1/N1 + 1/N2))
print(f'f-критерий для 95% доверительного интервала: {z:.4f}')

f-критерий для 95% доверительного интервала: -3.2765


Критическое значение z-критерия при α=5% и двухсторонней альтернативе равно 1,95996.  
-3.2765 < 1.96 , то есть расчетное значение z меньше критического значения.  
Нулевая гипотеза отвергается.  
Внедрение упрощенной формы заказа в магазине Утконос дало статистически значимый эффект на уровень конверсии в покупку, т.о. рекомендуется в дальнейшем использовать упрощенную форму оформления заказа.  

### Задание 2  
Сравниваем метрику конверсия в покупку. Размер выборки - 10000 элементов в каждой группе . Какой статистический критерий тут лучше всего подойдёт и почему?

In [ ]:
z-тест лучше подходит для задач с большими размерами выборки. Однако нужно учитывать ряд факторов: